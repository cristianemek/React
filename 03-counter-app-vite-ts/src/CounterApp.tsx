import { useState } from "react";


export const CounterApp = () => {
    const [value,setValue] = useState(0);
    const [counter, setCounter] = useState(value);

const handleIncrease = () => setCounter(counter + 1);
const handleDecrease = () => setCounter(counter - 1);
const handleReset = () => setCounter(value);

  return (
    <>
      <h1>Counter App</h1>
      <h2> {counter} </h2>
      <button onClick={() =>handleIncrease()}>+1</button>
      <button onClick={() => handleDecrease()}>-1</button>
      <button onClick={() =>handleReset()}>reset</button>
      <input 
      type="number"
      value={value}
      onChange={(e)=>{
        const newValue= Number(e.target.value);
        setValue(newValue);
        setCounter(newValue);
      }}
      />
    </>
  );
};

export default CounterApp;
