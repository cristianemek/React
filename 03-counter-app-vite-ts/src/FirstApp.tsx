interface Props {
  tittle?: string;
  subtittle?: string;
}

export const FirstApp = ({tittle="titulo",subtittle}: Props )=> {
  return (
    <>
        <h1>{tittle}</h1>
      {/* <h1>{JSON.stringify(newMessage)}</h1> */}
      <p>{subtittle}</p>
    </>
  );
};

