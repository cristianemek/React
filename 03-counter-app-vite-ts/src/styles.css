html,body{
    background-color: #21232a;
    color: white;
    font-family: helvetica,arial,sans-serif;
    font-size: 1.3rem;
    padding: 70px;
    text-align: center;
}

button {
background-color: #4f8cff;
color: #fff;
border: none;
border-radius: 8px;
padding: 12px 28px;
font-size: 1rem;
font-weight: bold;
cursor: pointer;
box-shadow: 0 4px 14px rgba(79, 140, 255, 0.2);
transition: background 0.2s, transform 0.2s, box-shadow 0.2s;
outline: none;
margin: 8px 0;
width: 100px;
display: inline-flex;
justify-content: center;
margin-right: 12px;
}

button:hover {
    background-color: #2563eb;
    transform: translateY(-2px) scale(1.03);
    box-shadow: 0 6px 20px rgba(79, 140, 255, 0.3);
}

button:active {
    background-color: #1e40af;
    transform: translateY(1px) scale(0.98);
    box-shadow: 0 2px 8px rgba(79, 140, 255, 0.15);
}

input {
    background-color: #23272f;
    text-align: center;
    color: #fff;
    border: none;
    border-radius: 8px;
    padding: 12px 16px;
    font-size: 1rem;
    margin: 8px 0;
    width: 150px;
    box-shadow: 0 2px 8px rgba(79, 140, 255, 0.12);
    outline: none;
    transition: box-shadow 0.2s, background 0.2s;
}
input:focus {
    background-color: #283046;
    box-shadow: 0 4px 14px rgba(79, 140, 255, 0.25);
}