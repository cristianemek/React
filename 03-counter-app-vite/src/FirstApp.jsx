import PropTypes from 'prop-types';

export const FirstApp = ({tittle,subtittle}) => {

  return (
    <>
        <h1>{tittle}</h1>
      {/* <h1>{JSON.stringify(newMessage)}</h1> */}
      <p>{subtittle}</p>
    </>
  );
};




FirstApp.propTypes = {
  tittle: PropTypes.string.isRequired,
  subtittle: PropTypes.string.isRequired
};

FirstApp.defaultProps = {
  tittle: 'No hay titulo',
  subtittle: 'No hay subtitulo'
};
