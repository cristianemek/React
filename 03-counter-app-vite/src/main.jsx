import React from 'react';
import ReactDom from 'react-dom/client';


function App() {
  return (
    <div>
      <h1>Hola mundo!!!</h1>
      <p>This is a simple React application.</p>
    </div>
  );
}

ReactDom.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);