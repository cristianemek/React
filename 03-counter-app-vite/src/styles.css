html,body{
    background-color: #21232a;
    color: white;
    font-family: helvetica,arial,sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}