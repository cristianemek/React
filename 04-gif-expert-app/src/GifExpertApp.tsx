import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch','Dragon Ball']);

    const onAddCategory = (newCat :string) =>{

        if (categories.includes(newCat)) return;
        setCategories(cat =>[newCat,...cat]);
        
    }



  return (
      <>
        <div>GifExpertApp</div>

        <AddCategory 
        onNewValue={onAddCategory}
        />
            {

            categories.map(cat => ( 
                <GifGrid 
                    key={cat} 
                    category={cat} 
                /> 
        ))

            }



      </>
  )
}