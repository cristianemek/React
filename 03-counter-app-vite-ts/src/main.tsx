import React from 'react';
import ReactDom from 'react-dom/client';
// import { HelloWorldApp } from './HelloWorldApp.jsx';
// import { FirstApp } from './FirstApp.js';
import {CounterApp} from './CounterApp.tsx';
import './styles.css';


ReactDom.createRoot(document.getElementById('root')!).render(
  <React.StrictMode>
    {/* <FirstApp subtittle="soy un subtitulo"/> */}
    <CounterApp/>
  </React.StrictMode>
);