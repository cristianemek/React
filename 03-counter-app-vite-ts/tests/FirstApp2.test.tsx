import { render,screen } from "@testing-library/react";
import { FirstApp } from '../src/FirstApp';

    describe('pruebas en FirstApp2', () => { 

      const title='hola soy un titulo';


        test('debe de hacer match con el snapshot', () => { 

            const{container}= render(<FirstApp tittle={title}/> );
            expect(container).toMatchSnapshot();
         })

         test('should show message hola soy un titulo', () => { 

          render(<FirstApp tittle={title}/>);
          expect(screen.getByText(title)).toBeTruthy();


          })


          test('should show title in h1', () => {
          
            render(<FirstApp tittle={title}/>);
            expect(screen.getByRole('heading',{level: 1}).innerHTML).toContain(title);
           })





     })

