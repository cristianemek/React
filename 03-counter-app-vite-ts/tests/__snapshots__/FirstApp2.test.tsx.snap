// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en FirstApp2 debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    hola soy un titulo
  </h1>
  <p />
  <p />
</div>
`;
